client/src/App.js

```diff
@@ -3,26 +3,62 @@ import { Route } from "react-router-dom";
 import SavedList from "./Movies/SavedList";
 import MovieList from "./Movies/MovieList";
 import Movie from "./Movies/Movie";
-import axios from 'axios';
+import UpdateMovie from "./Movies/UpdateMovie";
+import axios from "axios";
+import { useParams } from "react-router-dom";
+
+const initialMovieState = {
+  title: "",
+  director: "",
+  metascore: "",
+  stars: [],
+};
 
 const App = () => {
   const [savedList, setSavedList] = useState([]);
   const [movieList, setMovieList] = useState([]);
+  const [movie, setMovie] = useState(initialMovieState);
+  const params = useParams();
 
   const getMovieList = () => {
     axios
       .get("http://localhost:5000/api/movies")
-      .then(res => setMovieList(res.data))
-      .catch(err => console.log(err.response));
+      .then((res) => setMovieList(res.data))
+      .catch((err) => console.log(err.response));
   };
 
-  const addToSavedList = movie => {
+  const addToSavedList = (movie) => {
     setSavedList([...savedList, movie]);
   };
 
+  const onEditSubmit = () => {
```

Comment:
de ce in edit submit sunt 2 fetch-uri?

Reply:
unul pentru get, si al doilea pentru put

Reply:
de ce la submit ai nevoie de get?

Reply:
de fapt nu la submit, dar cand apare componenta, trebuie sa se afiseza datele despre movie in forma

Reply:
inteleg de ce get nu trebuie sa fie in aceasta functie, dar ma gandesc unde sa-l scriu, pentru ca in MovieForm la sigur nu poate fi scris

Reply:
de ce sa nu il poti scrie in MovieForm?

Reply:
ok, inseamna ca trebuie sa pun o conditie in dependenta de care acest get sa se execute. de exemplu daca props.history.path ="edit-movie"??

Reply:
da, doar ca eu as prefera sa fie in baza de propuri. Sa-i spunem ca din app pentru edit form ii dai un prop: populateWithData={true}, astfel deja in componenta form o sa faci fetch in baza de conditia ceea

Reply:
ok, am inteles. inca nu ma pot deprinde cu faptul ca actiunile tot pot fi state-uri, gen refresh, populateWithData

Reply:
gen props-uri?

Reply:
da

Reply:
nu ,adica cu propsurile totul e clar

Reply:
noi doar am avut refresh, setRefresh ca state

Reply:
la asta ma refeream

Reply:
poti sa-mi spui mai multe la ce nu e clar?

Reply:
de exemplu, cand zic state, prima definitie care imi vine este ca state-ul contine niste date, de ex items, movies, friends. Dar refresh de ex, este o actiune- a face refresh la pagina care poate fi true sau false. si anume imi vine greu sa asociez o actiune cu state-ul. intelegeti ?

Reply:
ah, okay, da cred ca ar trebui sa intelegem ca state-ul poate fi orice, chiar si o functie daca dorim

Reply:
Ok, cred ca depinde deja de contextul aplicatiei